openapi.yaml

```diff
@@ -4711,16 +4711,24 @@ components:
       example: null
       description: |-
         The namespace the `process_id` is valid for.
+
+        The following options are pre-defined by the openEO API, but additional
+        namespaces may be introduced by back-ends or in a future version of the API.
         
-        By default, the namespace is `null`, which includes all pre-defined processes listed at `GET /processes`.
-        The namespace `user` refers to all user-defined processes listed as `GET /process_graphs`.
-        Additional namespaces may be introduced by back-ends or in a future version of the API.
+        * `null` (default): Checks both user-defined and pre-defined processes,
+           but prefers user-defined processes if both are available.
+           This allows users to replace missing pre-defined processes with
+           user-defined processes, e.g. processes from https://processes.openeo.org
+           that have a process graph included.
+        * `backend`: Uses exclusively the pre-defined processes listed at `GET /processes`.
+        * `user`: Uses exclusively the user-defined processes listed at  listed as `GET /process_graphs`.  
     process_id:
       type: string
       description: |-
         Unique identifier for the process.
         
         MUST be unique across its namespace (e.g. pre-defined processes or user-defined processes).
+        Additionally, user-defined processes with an identifier equal to any pre-defined process MUST be rejected.
```

Comment:
Why is this necessary? It implies that namespaces are not independent, which breaks the point of namespaces.

Reply:
Just to avoid confusion. Maybe users don't realize there's a process with the name and then with the default behavior in PGs it seems a bit less error-prone to just reject them. We could make this a recommendation only?!

Reply:
It seems unnecessary to me because it's possible to refer to either one with the namespace property, so there is no unavoidable collision.
Also: even if you make it a recommendation in the API, a backend still has to make a hard choice what to do with it: accept  or reject (there is no channel for warnings or something like that)

Reply:
Has been replaced by a "clients should warn".